service/src/SecurePath.cpp

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2015 - 2023, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include "config.h"
+
+#include "SecurePath.hpp"
+
+#include <fcntl.h>
+#include <unistd.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+
+#include <string>
+#include <iostream>
+#include <sstream>
+
+#include "geopm/Exception.hpp"
+#include "geopm/Helper.hpp"
+
+namespace geopm
+{
+    SecurePath::SecurePath(const std::string &orig_path)
+        : SecurePath(orig_path,
+                     (S_IWGRP | S_IWOTH), // 0o022
+                     (getuid() == 0))
+    {
+
+    }
+
+    SecurePath::SecurePath(const std::string &orig_path, mode_t umask, bool enforce)
+        : m_fd(open(orig_path.c_str(), O_RDONLY))
+    {
+        if (m_fd < 0) {
+            throw Exception("SecurePath::" + std::string(__func__) +
+                            "(): Failed to open file: ",
+                            errno ? errno : GEOPM_ERROR_RUNTIME, __FILE__, __LINE__);
+        }
+
+        if (enforce) {
+            struct stat stat_struct;
+            int err = fstat(m_fd, &stat_struct);
+            if (err) {
+                (void)close(m_fd);
```

Comment:
```suggestion
                (void)!close(m_fd);
```
We need to put the ! in there with all these ignored return codes now (thanks Clang)